lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -1324,7 +1324,7 @@ SubExprInfo ExprEmitter::visitComb(MuxOp op) {
 }
 
 SubExprInfo ExprEmitter::visitTypeOp(StructCreateOp op) {
-  StructType stype = op.getType();
+  StructType stype = type_cast<StructType>(op.getType());
```

Comment:
Is there any way to get `StructCreateOp` to return a specific type?  We shouldn't have to cast on every user.

Reply:
Good point--yes. This was due to my desire to have all the places where `StructType` was previously used in ODS now accept the wrapper that denotes "`StructType`, or something that canonicalizes to `StructType`". I made that change across the board, but didn't think about the "create" ops. In reality, `StructCreateOp` will always create a `Value` of type `StructType` (at least at the moment). We can reflect this in ODS so `getType` returns the raw `StructType`, instead of the wrapper. The same holds for `ArrayCreateOp` and `UnionCreateOp`. I've made the change to the ODS return types here: https://github.com/llvm/circt/commit/862f1b4283c87917d668a3b88282a6b6327df68e

Reply:
Thank you!